coresdk/src/coresdk/database.h

```diff
@@ -1,60 +1,301 @@
+ *
+ * @returns Returns a `bool` which represents if there was a valid row of data to move to.
+ *
+ * @attribute class   database
+ * @attribute method  get_next_row
+ */
 bool get_next_row(query_result result);
 
+/**
+ * Checks if a given `query_result` has a valid row of data.
+ *
+ * @param result The `query_result` to check for a valid row of data on.
+ *
+ * @returns Returns a `bool` which represents if there was a valid row of data.
+ *
+ * @attribute class   database
+ * @attribute method  has_row
+ */
 bool has_row(query_result result);
 
+/**
+ * Resets a query_result back to its initial state, ready to be re-executed.
+ *
+ * @param result The `query_result` to reset.
+ *
+ * @attribute class   database
+ * @attribute method  reset_query_result
+ */
 void reset_query_result(query_result result);
 
+/**
+ * Queries a given column in the current row of the `query_result` for an integer.
+ *
+ * @param result The `query_result` to perform the query on.
+ *
+ * @param col The column you would like to query data from.
+ *
+ * @returns Returns the `int` which is the data at that column.
+ *
+ * @attribute class   database
+ * @attribute method  query_column_for_int
+ */
 int query_column_for_int(query_result result, int col);
 
+/**
+ * Queries a given column in the current row of the `query_result` for a double.
+ *
+ * @param result The `query_result` to perform the query on.
+ *
+ * @param col The column you would like to query data from.
+ *
+ * @returns Returns the `double` which is the data at that column.
+ *
+ * @attribute class   database
+ * @attribute method  query_column_for_double
+ */
 double query_column_for_double(query_result result, int col);
 
+/**
+ * Queries a given column in the current row of the `query_result` for a string.
+ *
+ * @param result The `query_result` to perform the query on.
+ *
+ * @param col The column you would like to query data from.
+ *
+ * @returns Returns the `string` which is the data at that column.
+ *
+ * @attribute class   database
+ * @attribute method  query_column_for_string
+ */
 string query_column_for_string(query_result result, int col);
 
+/**
+ * Queries a given column in the current row of the `query_result` for a bool.
+ *
+ * @param result The `query_result` to perform the query on.
+ *
+ * @param col The column you would like to query data from.
+ *
+ * @returns Returns the `bool` which is the data at that column.
+ *
+ * @attribute class   database
+ * @attribute method  query_column_for_bool
+ */
 bool query_column_for_bool(query_result result, int col);
 
+/**
+ * Queries a given column in the current row of the `query_result` for the data type at its postition.
+ *
+ * @param result The `query_result` to perform the query on.
+ *
+ * @param col The column you would like to know the data type of
+ *
+ * @returns Returns a `string` which is the name of the data type being stored.
+ * - Coule return: "INTEGER", "FLOAT", "TEXT", "NULL".
```

Comment:
Use `INTEGER` for better markdown parsing (i.e., with backticks). No need for list here unless we add descriptions of each:

```
Could return:

- `INTEGER` - An [integer column type](http://some.sql.website.to.define.types/integer)

- `FLOAT` - A [float column type](](http://some.sql.website.to.define.types/float)

...
```